dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -2324,9 +2334,10 @@ RtpsUdpDataLink::bundle_mapped_meta_submessages(
         switch (res.sm_._d()) {
           case HEARTBEAT: {
             const EntityId_t id = res.sm_.heartbeat_sm().writerId;
-            unique = counts.heartbeat_counts_[id].insert(res.sm_.heartbeat_sm().count.value).second;
-            OPENDDS_ASSERT(unique);
             result = helper.add_to_bundle(res.sm_.heartbeat_sm());
+            size_t bundle_index = result ? meta_submessage_bundles.size() - 1 : meta_submessage_bundles.size();
```

Comment:
Is `meta_submessage_bundles` guaranteed to be nonempty when `result` is true?

Reply:
Yes, since each new set of addresses considered will start a new bundle. See Line 2313.